src/modules/rlm_attr_filter/rlm_attr_filter.c

```diff
@@ -203,20 +206,21 @@ static rlm_rcode_t CC_HINT(nonnull(1,2)) attr_filter_common(void const *instance
 	for (pl = inst->attrs; pl; pl = pl->next) {
 		int fall_through = 0;
 		int relax_filter = inst->relaxed;
-
+		char *str;
+		fr_value_box_aprint(NULL, &str, pl->name, NULL);
 		/*
 		 *  If the current entry is NOT a default,
 		 *  AND the realm does NOT match the current entry,
 		 *  then skip to the next entry.
 		 */
-		if ((strcmp(pl->name, "DEFAULT") != 0) &&
-		    (strcmp(keyname, pl->name) != 0))  {
+		if ((strcmp(str, "DEFAULT") != 0) &&
```

Comment:
I'd simplify this to something like:

```
if (pl->name->vb_type != FR_TYPE_STRING) continue;
if ((strcmp(pl->name->vb_strvalue, "DEFAULT") !=0) &&
     (strcmp(keyname, pl->name->vb_strvalue) != 0)) {
    continue;
}
```

code profiling shows that `malloc / free` are huge overheads, so we try to avoid them where possible.